Assets/Scripts/MazeGenerator.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Collections;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeGenerator : MonoBehaviour
+{
+    [SerializeField]
+    private int width;
+
+    [SerializeField]
+    private int height;
+
+    public Transform brick;
+    private int[,] Maze;
```

Comment:
Why wouldn't maze which is constructed with integers that represent the tile type enumeration not be of type TileType?